lib/ask/runtime/questionnaire_ivr_simulator.ex

```diff
@@ -110,8 +110,12 @@ defmodule Ask.Runtime.QuestionnaireIvrSimulator do
     end
 
     sync_simulation = fn simulation, response ->
-      {simulation, reply} = handle_at_message(simulation, response)
-      QuestionnaireSimulator.sync_simulation(simulation, reply, "ivr", handle_app_reply)
+      case handle_at_message(simulation, response) do
```

Comment:
Maybe using `handle_at_message` isn't the best solution here. Maybe just checking `if response == "hangup"` would suffice?

Reply:
If you ask this because of [this conversation](https://github.com/instedd/surveda/pull/2069#discussion_r858751321), considering the type is String, I agree with you [here](https://github.com/instedd/surveda/pull/2069#discussion_r858946573), the current version of the code is the best Elixir way of expressing this conditional. I changed my mind. I wouldn't change it 🙂

Reply:
On the other hand, I'm not sure if "hangup" is an at_message. Maybe you asked because you think it isn't? Are `response` and `at_message` different concepts? I like that question! 🙃

Reply:
Sorry! I was reading an old version of your code 🤦 . I'll review it again!

Reply:
Something like this ?

```elixir
if response == "hangup" do
  QuestionnaireSimulator.stop_simulation(simulation, handle_app_reply)
else
  {simulation, reply} = handle_at_message(simulation, response)
  QuestionnaireSimulator.sync_simulation(simulation, reply, "ivr", handle_app_reply)
end
```

For me, yes, I read it easier. But it's your call, I think we shouldn't spend too much more time here. Both versions are good enough for me. Thanks for asking!